core/common/src/main/java/alluxio/Configuration.java

```diff
@@ -604,5 +623,44 @@ public static void validate() {
     checkTieredLocality();
   }
 
+  /**
+   * @return a list of worker-related configurations
+   */
+  public static List<ConfigProperty> getWorkerConfiguration() {
+    List<ConfigProperty> configInfoList = new ArrayList<>();
+    for (Map.Entry<String, String> entry : toMap().entrySet()) {
+      String keyName = entry.getKey();
+      if (isWorkerRelated(keyName)) {
+        Source source = getSource(PropertyKey.fromString(keyName));
+        String sourceStr;
+        if (source == Source.SITE_PROPERTY) {
+          sourceStr =
+              String.format("%s (%s)", source.name(), getSitePropertiesFile());
+        } else {
+          sourceStr = source.name();
+        }
+        configInfoList.add(new ConfigProperty()
+            .setName(keyName).setValue(entry.getValue()).setSource(sourceStr));
+      }
+    }
+    return configInfoList;
+  }
+
+  /**
+   * Checks if a PropertyKey is related to worker.
+   *
+   * @param keyName the name of the PropertyKey to check
+   * @return if the PropertyKey is worker related
+   */
+  private static boolean isWorkerRelated(String keyName) {
```

Comment:
Why can't we just use the worker scope for this type of check?

Reply:
I think if you use `Scope#contains()`, that should be what you want. Ideally, just checking for `Scope#contains(WORKER)` should be all that we need to do.